client/src/pages/dashboard/Dashboard.js

```diff
@@ -90,8 +86,32 @@ class Dashboard extends Component {
       });
   };
 
-  handleUpVote = (event, projectId) => {
+  hasUserVotedOnThisProject = (event, projectId, voteType) => {
```

Comment:
This function checks to see if the user has voted on this project in the past. It makes an API call and finds if this projectId exists in the users database. If it does, the user will be notified they cannot vote on it, and it ends. If the projectId is not in the users database, it will determine if this is an upvote or downvote request via switch statement. Depending on which vote, the program will execute either this.handleUpVote or this.handleDownVote.